models/demos/utils/llm_demo_utils.py

```diff
@@ -0,0 +1,114 @@
+    """
+    Create a benchmark data object and populate the object with the given measurements.
+
+    Pre-requisites:
+    - The measurements dictionary should contain the following keys: "compile_prefill", "compile_decode", "prefill_t/s", "decode_t/s", "decode_t/s/u"
+    - The profiler object should contain the start and end times for the steps "compile_prefill", "compile_decode", "inference_prefill", "inference_decode"
+
+    Optional (should be provided if measuring perf, not required for token verification):
+    - The targets dictionary should contain the following keys: "prefill_t/s", "decode_t/s", "decode_t/s/u"
+    - The N_warmup_iter dictionary should contain the following keys: "inference_prefill", "inference_decode"
+    """
+
+    assert all(
+        key in measurements
+        for key in [
+            "compile_prefill",
+            "compile_decode",
+            "prefill_t/s",
+            "decode_t/s",
+            "decode_t/s/u",
+        ]
+    )
+
+    benchmark_data = BenchmarkData()
+
+    # Add measurement data
+    benchmark_data.add_measurement(profiler, 0, "compile_prefill", "time(s)", measurements["compile_prefill"])
+    benchmark_data.add_measurement(profiler, 0, "compile_decode", "time(s)", measurements["compile_decode"])
+    benchmark_data.add_measurement(
+        profiler,
+        0,
+        "inference_prefill",
+        "tokens/s",
+        measurements["prefill_t/s"],
+        step_warm_up_num_iterations=N_warmup_iter["inference_prefill"]
+        if "inference_prefill" in N_warmup_iter
+        else None,
+        target=targets["prefill_t/s"] if "prefill_t/s" in targets else None,
+    )
+    benchmark_data.add_measurement(
+        profiler,
+        0,
+        "inference_decode",
+        "tokens/s",
+        measurements["decode_t/s"],
+        step_warm_up_num_iterations=N_warmup_iter["inference_decode"] if "inference_decode" in N_warmup_iter else None,
+        target=targets["decode_t/s"] if "decode_t/s" in targets else None,
+    )
+    benchmark_data.add_measurement(
+        profiler,
+        0,
+        "inference_decode",
+        "tokens/s/user",
+        measurements["decode_t/s/u"],
+        step_warm_up_num_iterations=N_warmup_iter["inference_decode"] if "inference_decode" in N_warmup_iter else None,
+        target=targets["decode_t/s/u"] if "decode_t/s/u" in targets else None,
+    )
+    return benchmark_data
+
+
+def verify_tokens(generated_text: dict, expected_greedy_output_path: str):
+    with open(expected_greedy_output_path, "r") as f:
+        expected_output = json.load(f)
+    does_pass = generated_text == expected_output
+    if does_pass:
+        logger.info("Output Check Passed!")
+    else:
+        assert (
+            does_pass
+        ), f"Generated text does not match expected output! \n\n Generated text:\n {generated_text} \n\n Expected output:\n {expected_output}"
+
+
+def verify_perf(measurements: dict, expected_perf_metrics: dict):
```

Comment:
Are you planning to make more generic versions of these verify/generate functions?

Reply:
No, the verify functions here are only intended to reduce duplicate code in the demos and are not related to the csvs. However, in the future we should rely on the targets in the csvs for asserting on all the tests so that we don't assert in the middle of the pipeline. That can be addressed in a separate PR after this one.